src/Logger.jl

```diff
@@ -111,3 +111,73 @@ function reset_tb_logs()
     logdir = default_logging_session[].logdir
     default_logging_session[] = Logger(logdir, overwrite=true)
 end
+
+
+# Implement the AbstractLogger Interface
+
+catch_exceptions(lg::TBLogger) = false
+
+min_enabled_level(lg::TBLogger) = lg.min_level
+
+# For now, log everything that is above the lg.min_level
+shouldlog(lg::TBLogger, level, _module, group, id) = true
+
+function handle_message(lg::TBLogger, level, message, _module, group, id, file, line; kwargs...)
+    # Unpack the message
+    summ    = SummaryCollection()
+    i_step = 1
+
+    if !isempty(kwargs)
+        for (key,val) in pairs(kwargs)
+            # special values
+            if key == :log_step_increment
+                i_step = val
+                continue
+            end
+
+            data = Stack{Pair{String,Any}}()
+            name = message*"/$key"
+            push!(data, name => val)
+            while !isempty(data)
+                name, val = pop!(data)
+                loggable(val) ? push!(summ.value, summary_impl(name, val)) : preprocess(name, val, data)
+            end
+        end
+    end
+    iter = increment_step(lg, i_step)
+    write_event(lg.file, make_event(lg, summ, step=iter))
+end
+
+"""
+    loggable(value) -> Bool
+
+Returns `true` if `value` is a type that can be serialized into a `Summary`
+ProtoBuffer, `false` otherwise.
+
+This is defined to be false for `::Any`, and for every supported TensorBoard
+Plugin this method should be specialized for the relative type and return true.
+"""
+loggable(::Any) = false
+
+"""
+    preprocess(name, val, data)
+
+This method takes a tag `name` and the value `val::T` which cannot be directly
+serialized into TensorBoard, and pushes into the stack `data` several
+name-value pairs `Pair{String,Any}` containing simpler types. Those pairs will
+be serialized if possible, otherwise `preprocess` will be called recursively.
+
+For a struct
+"""
+function preprocess(name, val::T, data) where T
```

Comment:
Shouldn't `preprocess`  be recursive?

Reply:
Nope: my idea was that `preprocess` would split things into _simpler_ things. The recursive nature was obtained at the call point as 
```
while !isempty(data)
   name, val = pop!(data)
   loggable(val) ? push!(summ.value, summary_impl(name, val)) : preprocess(name, val, data)
end
````

I don't remember why I had done this. I agree that it's not super-transparent, and a better approach would be to simply make `preprocess` recursive and specialize it for types that can be logged. This has the added benefit that we can loose the `loggable` function and the code becomes simpler.

Reply:
Oh, I see it is recursive because it mutates `data`.
Yeah, that is confusing.

Reply:
Ok. Implemented with the last commit.
Thanks for pointing this out.

Reply:
After a bit more thought
Preprocess should be call
```
decompose!(store, name, val:T
```

Were `store` takes the place of `data` as a slightly less generic name